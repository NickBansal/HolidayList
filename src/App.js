import React, { Component } from 'react';
import './Stylesheets/App.css'
import background from './Stylesheets/Images/background.jpg'
import holidayData from './Data'
import HolidayInformation from './Components/HolidayInformation'
import { alphaSort } from './utils'

class App extends Component {

  state = {
    holidayData,
    toggleReverse: false,
    sortValue: 'title'  
  }

  render() {
    const style = {
      backgroundImage: `url(${background})`
    }
    return (
      <div style={style} className='App'>
        <div className="SortButtons">
          <button value='title' onClick={(e) => this.handleClick(e.target.value)}>sort alphabetically</button>
          <button value='price' onClick={(e) => this.handleClick(e.target.value)}>sort by price</button>
          <button value='rating' onClick={(e) => this.handleClick(e.target.value)}>sort by star rating</button>
        </div>

        <HolidayInformation 
        toggleReverse={this.state.toggleReverse}
        holidayData={this.state.holidayData} />
      </div>
    );
  }

  handleClick = value => {
    const { holidayData, sortValue, toggleReverse } = this.state
    const newToggleReverse = sortValue !== value ? false : !toggleReverse

    const filteredHolidayData = value === 'title' ? 
    holidayData.sort(alphaSort) : 
    holidayData.sort((a, b) => b[value] - a[value])

    this.setState({
      holidayData: filteredHolidayData,
      toggleReverse: newToggleReverse,
      sortValue: value
    })
  }
}

export default App;
