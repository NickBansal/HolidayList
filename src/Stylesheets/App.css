.App {
  text-align: center;
  padding: 40px;
  display: flex;
  flex-direction: column;
  align-items: center
}

.App h1 {
  margin: 0
}

.HolidayInformation {
  margin: 20px;
  position: relative;
  height: 300px;
  width: 400px;
}

.HolidayInformation img {
  height: 100%;
  width: 100%;
} 

.HoldiayInformationTitle {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  background: #add8e6d9;
}