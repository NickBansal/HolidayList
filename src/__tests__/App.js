import '../setupTest'
import React from 'react'
import App from '../App'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('<App />', () => {

  let wrapper 
  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  it('Component matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Changes the sortValue in state on click handler', () => {

    wrapper.instance().handleClick('title')
    expect(wrapper.state().sortValue).toBe('title')
    wrapper.instance().handleClick('price')
    expect(wrapper.state().sortValue).toBe('price')
    wrapper.instance().handleClick('ratings')
    expect(wrapper.state().sortValue).toBe('ratings')
  })
  it('Changes the toggleReverse in state on click handler', () => {

    wrapper.instance().handleClick('title')
    expect(wrapper.state().toggleReverse).toBe(false)
    wrapper.instance().handleClick('title')
    expect(wrapper.state().toggleReverse).toBe(true)
    wrapper.instance().handleClick('title')
    expect(wrapper.state().toggleReverse).toBe(false)
    wrapper.instance().handleClick('price')
    expect(wrapper.state().toggleReverse).toBe(false)
 
  })
  it('Changes the showDescription in state on click handler', () => {
    wrapper.instance().toggleDescription(1)
    expect(wrapper.state().showDescription).toBe(true)
    expect(wrapper.state().holidayElement).toBe(1)
    wrapper.instance().toggleDescription(2)
    expect(wrapper.state().showDescription).toBe(true)
    expect(wrapper.state().holidayElement).toBe(2)
    wrapper.instance().toggleDescription(3)
    expect(wrapper.state().showDescription).toBe(true)
    expect(wrapper.state().holidayElement).toBe(3)
    wrapper.instance().toggleDescription(3)
    expect(wrapper.state().showDescription).toBe(false)
    expect(wrapper.state().holidayElement).toBe(3)
 
  })
})
