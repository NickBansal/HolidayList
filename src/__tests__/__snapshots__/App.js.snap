// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component matches the snapshot 1`] = `
<div
  className="App"
  style={
    Object {
      "backgroundImage": "url(background.jpg)",
    }
  }
>
  <SortButtons
    handleClick={[Function]}
    sortByElement={1}
  />
  <HolidayInformation
    holidayData={
      Array [
        Object {
          "airport": "East Midlands",
          "date": "3 July 2014",
          "days": "7 days",
          "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
          "img": "image_01.png",
          "location": "Costa Adeje, Tenerife",
          "price": "1136.50",
          "rating": "5",
          "specification": "2 Adults, 2 Children & 1 infant",
          "title": "Costa Adeje Gran Hotel",
        },
        Object {
          "airport": "Liverpool",
          "date": "27 May 2014",
          "days": "7 days",
          "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
          "img": "image_03.jpg",
          "location": "Costa Adeje, Tenerife",
          "price": "696.80",
          "rating": "3",
          "specification": "2 Adults, 1 Child",
          "title": "Laguna Park II",
        },
        Object {
          "airport": "Manchester",
          "date": "3 July 2014",
          "days": "7 days",
          "description": "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.",
          "img": "image_02.png",
          "location": "Costa Adeje, Tenerife",
          "price": "499.99",
          "rating": "4",
          "specification": "2 Adults, 2 Children",
          "title": "Aguamarina Golf Hotel And Apartments",
        },
      ]
    }
    showDescription={false}
    toggleDescription={[Function]}
    toggleReverse={false}
  />
</div>
`;
